github3/repos/repo.py

```diff
@@ -524,17 +524,24 @@ def code_frequency(self, number=-1, etag=None):
         url = self._build_url('stats', 'code_frequency', base_url=self._api)
         return self._iter(int(number), url, list, etag=etag)
 
-    def collaborators(self, number=-1, etag=None):
+    def collaborators(self, affiliation='all', number=-1, etag=None):
         r"""Iterate over the collaborators of this repository.
 
         :param int number: (optional), number of collaborators to return.
             Default: -1 returns all comments
         :param str etag: (optional), ETag from a previous request to the same
             endpoint
-        :returns: generator of :class:`~github3.users.ShortUser`\ s
+        :returns: generator of :class:`~github3.repos.users.Collaborator`\ s
         """
         url = self._build_url('collaborators', base_url=self._api)
-        return self._iter(int(number), url, users.ShortUser, etag=etag)
+        affiliations = {'outside', 'direct', 'all'}
+        if affiliation not in affiliations:
+            raise ValueError(
+                "Invalid affiliation parameter passed, must be 'outside', "
```

Comment:
I wonder if we should format the given affiliation into this to make it easier for debugging. Perhaps:

```py
("Invalid affiliation value {!r} parameter passed, must be 'outside',"
 "'direct', or 'all' (defaults to 'all').").format(affiliation)
```